File: Factura.js
function factura() {
	var nombre = new String();
	var direccion = new String();
	var continua = new String();
	var cliente = new String();
	var descripcion = new String();
	var cantidad = new Number();
	var nro_art = new Number();
	var fila = new Number();
	var cf = new Number();
	var codigo = new Number();
	var cantidad_venta = new Number();
	var articulo = new Number();
	var i = new Number();
	var j = new Number();
	var pre_unit = new Number();
	var total = new Number();
	var subtotal = new Number();
	var encuentra = new Boolean();
	// Array para guardar los datos de la factura;
	var articulo = new Array(3);
	var cantidad = new Array(3);
	var pre_unit = new Array(3);
	var descripcion = new Array(3);
	document.write(" ",'<BR/>');
	document.write("Ingresa articulos?",'<BR/>');
	document.write("=================",'<BR/>');
	continua = "s";
	nro_art = 1;
	total = 0;
	// Ingreso de articulos al arreglo
	do {
		document.write("Artículo Número ",nro_art,":");
		articulo[nro_art-1] = Number(prompt("Ingrese Articulo <nro entero>"));
		document.write("cantidad: ");
		cantidad[nro_art-1] = Number(prompt("Ingrese cantidad"));
		document.write("descripcion: ");
		descripcion[nro_art-1] = prompt("Ingrese descripcion");
		document.write("Precio unitario: ");
		pre_unit[nro_art-1] = Number(prompt("Ingrese Precio unitario"));
		nro_art = nro_art+1;
		document.write("Ingresa otro articulo? (s/n)",'<BR/>');
		continua = prompt("Ingresa otro articulo? (s/n)");
		document.write(" ",'<BR/>');
	} while (continua!="n");
	nro_art = nro_art-1;
	document.write(" ",'<BR/>');
	document.write("Ingrese datos del cliente",'<BR/>');
	document.write("=========================",'<BR/>');
	document.write("Nombre: ");
	nombre = prompt("Nombre: ");
	document.write("Dirección: ");
	direccion = prompt("Direccion: ");
	document.write(" ",'<BR/>');
	document.write(" ",'<BR/>');
	continua = "s";
	cf = 1;
	cliente = "s";
	encuentra = false;
	j = 0;
	do {
		document.write("Factura N° 00000 - ",cf,'<BR/>');
		document.write("Cantidad       Descripcion       Precio Unitario       Subtotal",'<BR/>');
		document.write("_______________________________________________________________",'<BR/>');
		do {
			document.write("Ingrese codigo del producto: ",'<BR/>');
			codigo = Number(prompt("Ingrese codigo del producto: "));
			document.write("cantidad en la venta: ",'<BR/>');
			cantidad_venta = Number(prompt("cantidad en la venta: "));
			// buscar codigo en vector
			for (i=1;i<=nro_art;i++) {
				if (articulo[i-1]==codigo) {
					encuentra = true;
					j = i;
				}
			}
			if (encuentra) {
				subtotal = cantidad_venta*pre_unit[j-1];
				total = total+subtotal;
				document.write(cantidad_venta,"            ",descripcion[j-1],"             ",pre_unit[j-1],"          ",subtotal,'<BR/>');
				// Mostrar factura
				encuentra = false;
			} else {
				// Escribir error en la busqueda
				document.write(" ",'<BR/>');
				document.write("******************",'<BR/>');
				document.write("codigo inexistente",'<BR/>');
				document.write("******************",'<BR/>');
				document.write(" ",'<BR/>');
			}
			document.write("Hay mas articulos en el carrito? (s/n)",'<BR/>');
			continua = prompt("Hay mas articulos en el carrito? (s/n)");
		} while (continua!="n");
		document.write("Total factura ",total,'<BR/>');
		total = 0;
		continua = "s";
		cf = cf+1;
		document.write("Hay otro cliente? (s/n)",'<BR/>');
		cliente = prompt("Hay otro cliente? (s/n)");
	} while (cliente!="n");
	document.write("PRESIONE F5 - PARA CONTINUAR",'<BR/>');
}

File: Factura.java
/* Este codigo ha sido generado por el modulo psexport 20180802-w32 de PSeInt.
Es posible que el codigo generado no sea completamente correcto. Si encuentra
errores por favor reportelos en el foro (http://pseint.sourceforge.net). */

// En java, el nombre de un archivo fuente debe coincidir con el nombre de la clase que contiene,
// por lo que este archivo debería llamarse "FACTURA.java."

import java.io.*;

public class factura {

	public static void main(String args[]) throws IOException {
		BufferedReader bufEntrada = new BufferedReader(new InputStreamReader(System.in));
		int articulo[];
		int cantidad[];
		int cantidad_venta;
		int cf;
		String cliente;
		int codigo;
		String continua;
		String descripcion[];
		String direccion;
		boolean encuentra;
		int fila;
		int i;
		int j;
		String nombre;
		int nro_art;
		double pre_unit[];
		double subtotal;
		double total;
		// Array para guardar los datos de la factura;
		articulo = new int[3];
		cantidad = new int[3];
		pre_unit = new double[3];
		descripcion = new String[3];
		System.out.println(" ");
		System.out.println("Ingresa articulos?");
		System.out.println("=================");
		continua = "s";
		nro_art = 1;
		total = 0;
		do {
			System.out.print("Artículo Número "+nro_art+":");
			articulo[nro_art-1] = Integer.parseInt(bufEntrada.readLine());
			System.out.print("cantidad: ");
			cantidad[nro_art-1] = Integer.parseInt(bufEntrada.readLine());
			System.out.print("descripcion: ");
			descripcion[nro_art-1] = bufEntrada.readLine();
			System.out.print("Precio unitario: ");
			pre_unit[nro_art-1] = Double.parseDouble(bufEntrada.readLine());
			nro_art = nro_art+1;
			System.out.println("Ingresa otro artículo? (s/n)");
			continua = bufEntrada.readLine();
			System.out.println(" ");
		} while (!continua.equals("n"));
		nro_art = nro_art-1;
		System.out.println(" ");
		System.out.println("Ingrese datos del cliente");
		System.out.println("=========================");
		System.out.print("Nombre: ");
		nombre = bufEntrada.readLine();
		System.out.print("Dirección: ");
		direccion = bufEntrada.readLine();
		System.out.println(" ");
		// Escribir "Ingresa articulos?";
		// Escribir "=================";
		// continua <- "s";
		// nro_art <- 1;
		// total <- 0;
		// Repetir
		// Escribir Sin Saltar "Artículo Número ", nro_art, ":";
		// Leer articulo[nro_art];
		// Escribir Sin Saltar "cantidad: ";
		// Leer cantidad[nro_art];
		// Escribir Sin Saltar "descripcion: ";
		// Leer descripcion[nro_art];
		// Escribir Sin Saltar "Precio unitario: ";
		// Leer pre_unit[nro_art];
		// nro_art <- nro_art + 1;
		// Escribir "Ingresa otro artículo? (s/n)";
		// Leer continua;
		// Escribir " ";
		// Hasta Que continua = "n"
		// nro_art <- nro_art - 1;
		System.out.println(" ");
		continua = "s";
		cf = 1;
		cliente = "s";
		encuentra = false;
		j = 0;
		do {
			System.out.println("Factura N° 00000 - "+cf);
			System.out.println("Cantidad     Descripcion     Precio Unitario     Subtotal");
			System.out.println("_________________________________________________________");
			do {
				System.out.println("Ingrese código del producto: ");
				codigo = Integer.parseInt(bufEntrada.readLine());
				System.out.println("cantidad en la venta: ");
				cantidad_venta = Integer.parseInt(bufEntrada.readLine());
				// buscar codigo en vector
				for (i=1;i<=nro_art;i++) {
					if (articulo[i-1]==codigo) {
						encuentra = true;
						j = i;
					}
				}
				if (encuentra) {
					subtotal = cantidad_venta*pre_unit[j-1];
					total = total+subtotal;
					System.out.println(cantidad_venta+"        "+descripcion[j-1]+"        "+pre_unit[j-1]+"        "+subtotal);
					// Mostrar factura
					encuentra = false;
				} else {
					// Escribir error en la busqueda
					System.out.println(" ");
					System.out.println("******************");
					System.out.println("código inexistente");
					System.out.println("******************");
					System.out.println(" ");
				}
				System.out.println("Hay más artículos en el carrito? (s/n)");
				continua = bufEntrada.readLine();
			} while (!continua.equals("n"));
			System.out.println("Total factura "+total);
			total = 0;
			continua = "s";
			cf = cf+1;
			System.out.println("Hay otro cliente? (s/n)");
			cliente = bufEntrada.readLine();
		} while (!cliente.equals("n"));
	}


}

